File: scripts/verify_integration.py
#!/usr/bin/env python3
"""Same checks as verify-integration.mjs — uses only stdlib (no Node required)."""
from __future__ import annotations

import os
import sys
import urllib.error
import urllib.request

BASE = os.environ.get("VERIFY_API_URL", "http://127.0.0.1:8000").rstrip("/")


def main() -> None:
    health = urllib.request.urlopen(f"{BASE}/health/", timeout=30)
    body = health.read().decode().strip()
    if health.status != 200 or body != "ok":
        raise SystemExit(f"GET /health/ expected 200 and body 'ok', got {health.status} {body!r}")

    schema = urllib.request.urlopen(f"{BASE}/api/schema/", timeout=30)
    if schema.status != 200:
        raise SystemExit(f"GET /api/schema/ expected 200, got {schema.status}")

    print(f"Integration OK — {BASE}")


if __name__ == "__main__":
    try:
        main()
    except urllib.error.URLError as e:
        print(e, file=sys.stderr)
        print(
            "\nStart Django first: Backend/ → pip install -r requirements.txt && "
            "python manage.py runserver 127.0.0.1:8000\n",
            file=sys.stderr,
        )
        sys.exit(1)

File: scripts/verify-integration.mjs
#!/usr/bin/env node
/**
 * Smoke-check that the Django API is up and wired (health + OpenAPI schema).
 * Run with the backend already listening (local or deployed).
 *
 *   npm run verify
 *   VERIFY_API_URL=https://your-api.onrender.com npm run verify
 */
const base = (process.env.VERIFY_API_URL || 'http://127.0.0.1:8000').replace(/\/$/, '')

async function assertOk(res, label) {
  if (!res.ok) {
    const body = await res.text()
    throw new Error(`${label} → HTTP ${res.status} ${body.slice(0, 200)}`)
  }
}

async function main() {
  const h = await fetch(`${base}/health/`)
  await assertOk(h, 'GET /health/')
  const healthText = (await h.text()).trim()
  if (healthText !== 'ok') {
    throw new Error(`GET /health/ expected body "ok", got: ${healthText.slice(0, 100)}`)
  }

  const schema = await fetch(`${base}/api/schema/`)
  await assertOk(schema, 'GET /api/schema/')
  const ct = schema.headers.get('content-type') || ''
  if (!ct.includes('json') && !ct.includes('yaml')) {
    console.warn('Warning: /api/schema/ content-type unexpected:', ct)
  }

  console.log(`Integration OK — ${base}`)
}

main().catch((err) => {
  console.error(err.message || err)
  console.error(
    '\nStart the backend first, e.g. from Backend/: python -m pip install -r requirements.txt && python manage.py migrate && python manage.py runserver 127.0.0.1:8000\n'
  )
  process.exit(1)
})
